added rulesVC

MalinovIS opened the pull request:
added rulesVC

MalinovIS committed:
added rulesVC

pashaGriDev commented on MillionaireApp/UI/Start/StartViewController.swift:
```diff
@@ -110,6 +110,8 @@ final class StartViewController: UIViewController {
-        //TODO: destinationVC (rules of game)
+        let rulesViewController = RulesViewController()
+        self.navigationController?.pushViewController(rulesViewController, animated: true)
+        
```
Давай изменим презентацию правил. Сделай так чтоб правила показывались с низу вверх. метод `present` и стиль не на весь экран.

pashaGriDev commented on MillionaireApp/UI/Rules/RulesViewController.swift:
```diff
@@ -0,0 +1,132 @@
+
+    Позвоните другу - участникам разрешается сделать 30-секундный звонок другу или члену семьи и спросить, знают ли они ответ на вопрос.
+
+"""
```
Я не думаю что такой информации здесь место. Можешь перенести это в модель. Не проблема сделать небольшую структуру и потом сделать для нее расширения или метод в который все перенести. Так же можно разбить текст на несколько частей. Можно что-то выделить цветом.

pashaGriDev commented on MillionaireApp/UI/Rules/RulesViewController.swift:
```diff
@@ -0,0 +1,132 @@
+        label.textAlignment = .center
+        label.font = .systemFont(ofSize: 36, weight: .bold)
+        return label
+    }()
```
этот элемент ведет себя странно. Он больше похож на Нав бар. Я думаю будет лучше чтоб он мог скроллиться вместе с другим контентом.

pashaGriDev commented on MillionaireApp/UI/Rules/RulesViewController.swift:
```diff
@@ -0,0 +1,132 @@
+        let label = UILabel()
+        label.text = "Правила игры"
+        label.textColor = .white
+        label.numberOfLines = 0
```
для чего здесь это свойство?

pashaGriDev commented on MillionaireApp/UI/Rules/RulesViewController.swift:
```diff
@@ -0,0 +1,132 @@
+        button.titleLabel?.font = .systemFont(ofSize: 36, weight: .bold)
+        button.addTarget(self, action: #selector(quitButtonAction), for: .touchUpInside)
+        return button
+    }()
```
Если мы изменим презентацию экрана, то эта кнопка нам не понадобиться.

pashaGriDev commented on MillionaireApp/UI/Rules/RulesViewController.swift:
```diff
@@ -0,0 +1,132 @@
+        view.addSubviews([backgroundImageView, mainLabel,scrollView])
+        scrollView.addSubview(rulesLabel)
+        scrollView.addSubview(quitButton)
+        
```
Чем вам не нравиться метод который вы использовали на 94 строке и вы предпочли этоти?

pashaGriDev commented on MillionaireApp/UI/Rules/RulesViewController.swift:
```diff
@@ -0,0 +1,132 @@
+            make.top.equalTo(mainLabel.snp.bottom).offset(20)
+            make.leading.equalToSuperview()
+            make.trailing.equalToSuperview()
+            make.bottom.equalToSuperview()
```
совет: три строки выше можно написать одной строкой.

MalinovIS committed:
Changed Present(),
added Model, extension for formatted String,
replaced mainLabel to ScrollView,
deleted numberoflines in mainLabel, quitButton.

pashaGriDev commented on MillionaireApp/UI/Rules/RulesViewController.swift:
```diff
@@ -0,0 +1,88 @@
+        formattedText.append(.formattedText(rulesStruct.rulesSecondQuestionsDescribeText, attributes: [.foregroundColor: UIColor.white, .font: UIFont.boldSystemFont(ofSize: 12)]))
+        formattedText.append(.formattedText(rulesStruct.rulesThirdQuestionsText, attributes: [.foregroundColor: UIColor.green, .font: UIFont.boldSystemFont(ofSize: 12)]))
+        formattedText.append(.formattedText(rulesStruct.rulesTipsText, attributes: [.foregroundColor: UIColor.white, .font: UIFont.boldSystemFont(ofSize: 12)]))
+        rulesLabel.attributedText = formattedText
```
Почему бы вам не перенести все настройки formattedText в отдельный метод?
Я не думаю что правильно использовать Label для такого большого объема текста, но я не стану настаивать на переделке. Вы могли бы использовать например таблицу и разделить правила как вам захочется или разделить правила на несколько лайблов.

pashaGriDev commented on MillionaireApp/UI/Rules/RulesViewController.swift:
```diff
@@ -0,0 +1,88 @@
+            
+        }
+        
+        
```
удалите пожалуйста лишние пустые строки

pashaGriDev commented on MillionaireApp/UI/Rules/RulesViewController.swift:
```diff
@@ -0,0 +1,88 @@
+        
+        
+    }
+    
```
и тут

pashaGriDev commented on MillionaireApp/UI/Rules/RulesViewController.swift:
```diff
@@ -0,0 +1,88 @@
+        rulesLabel.attributedText = formattedText
+    }
+    
+    
```
тут тоже пустая строка 1 лишняя

pashaGriDev commented on MillionaireApp/UI/Rules/RulesViewController.swift:
```diff
@@ -0,0 +1,88 @@
+        rules.font = .systemFont(ofSize: 14, weight: .medium)
+        return rules
+    }()
+
```
пустая строка

pashaGriDev commented on MillionaireApp/Model/Rules.swift:
```diff
@@ -0,0 +1,56 @@
+// MillionaireApp, 2024
+
+import Foundation
+struct RulesStruct {
```
Пожалуйста переместите папку модель в вашу папку правила. Эта модель не относиться к общим моделям для этого приложения.

pashaGriDev commented on MillionaireApp/UI/Start/StartViewController.swift:
```diff
@@ -110,6 +110,8 @@ final class StartViewController: UIViewController {
     @objc private func rulesButtonAction() {
-        //TODO: destinationVC (rules of game)
+        let rulesViewController = RulesViewController()
+        present(rulesViewController, animated: true, completion: nil)
```
если удалить `, completion: nil` - ничего страшного не произойдет.

pashaGriDev commented on MillionaireApp/UI/Rules/RulesViewController.swift:
```diff
@@ -0,0 +1,88 @@
+    
+    private let rulesLabel: UILabel = {
+       let rules = UILabel()
+        rules.text = ""
```
для чего здесь `rules.text = ""` какую функцию выполняет?

pashaGriDev commented on MillionaireApp/UI/Rules/RulesViewController.swift:
```diff
@@ -0,0 +1,88 @@
+    private let scrollView: UIScrollView = {
+        let scrollView = UIScrollView()
+        return scrollView
+    }()
```
Почему это свойство не может быть в одну строку, как вы настраиваете его внутри кложуры?

MalinovIS committed:
Replaced Model to the Rules,
deleted empty lines,
added formattedRules

MalinovIS committed:
delete empty lines

pashaGriDev committed:
fix: delete folder Entity

pashaGriDev committed:
Merge branch 'develop' into feature/MA_RulesScreen

# Conflicts:
#	MillionaireApp.xcodeproj/project.pbxproj

pashaGriDev committed:
feat: connected coordinator
